### DISCUSSION TITLE
Arduino ESP32 core outdated

### DISCUSSION THREAD

**Discussion by Jason2866** [benzino77/tasmocompiler#229] [General]
Hello @benzino77 
Since Arduino ESP32 Core 2.0.1 is released and we build since weeks Tasmota32  version with beta version from this core and all seems working better as with the old core we switched to
our build based on this core. Since yesterday EVERY Tasmota version for ESP32 Mcu variants 
is builded with. Support for core 1.0.7.x is dropped. New upcoming features will not work with.
Today already code changes where done which are incompatible with the old core.
Please update your setup if you want to provide development Tasmota ESP32 builds.
Users do like your work a lot as we do too.
Greets from the Tasmota dev team.

Thanks! 

Jason2866

**Comment by benzino77** [comment]
Thanks for the info. 
I think that ESP32 boards based on `extends = env:tasmota32_base` will work as expected. I'm just switching to git tags before compilation so it will work with development branch and current stable 10.x.
This is how `platformio_override.ini` file looks like for particular ESP32 boards:
esp32generic:
```ini
[platformio]
default_envs = firmware

[env:firmware]
extends = env:tasmota32_base
```

esp32-webcam:
```ini
[platformio]
default_envs = firmware

[env:firmware]
extends = env:tasmota32_base
board = esp32-cam
```

odroid-go:
```ini
[platformio]
default_envs = firmware

[env:firmware]
extends = env:tasmota32_base
board = esp32-odroid
```

m5stack-core:
```ini
[platformio]
default_envs = firmware

[env:firmware]
extends = env:tasmota32_base
board = esp32-m5core2
```
It should rather work with current development version.

There will be a problem with Solo1 which (and this is a very good move) has started using additional level of abstraction: `platform_packages       = ${core32solo1.platform_packages}`
C3 and S2 boards will also probably have some problems because of `platform_packages` and other platformio entries has changed since latest stable.

I will rather leave it as is, to be compatible with latest stable version, but I will probably launch additional TasmoCompiler branch (let's say development) which will follow the latest changes in Tasmota code and be compatible with Tasmota development branch.

**Reply by Jason2866** [reply]
Hi!
Your planned changes LGTM, and should work. As far as i can say 😀
The setup changes for S2 and C3 should be (not tested) backwards compatible to release.
Greets Jason2866

**Comment by Jason2866** [comment]
Made a PR for updating some Board defs (Solo1 does not compile anymore).
